controllers/datadogagent/feature/admissioncontroller/feature_test.go

```diff
@@ -2,129 +2,126 @@
+	apmSocketHostPath  = apicommon.DogstatsdAPMSocketHostPath + "/" + apicommon.APMSocketName
+	apmSocketLocalPath = apicommon.APMSocketVolumeLocalPath + "/" + apicommon.APMSocketName
+	customPath         = "/custom/host/filepath.sock"
+)
 
+func Test_admissionControllerFeature_Configure(t *testing.T) {
 	tests := test.FeatureTestSuite{
-		//////////////////////////
-		// v1Alpha1.DatadogAgent
-		//////////////////////////
 		{
-			Name:          "v1alpha1 admission controller not enabled",
-			DDAv1:         newV1Agent(false),
+			Name: "v2alpha1 Admission Controller not enabled",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(false).
+				Build(),
 			WantConfigure: false,
 		},
 		{
-			Name:          "v1alpha1 admission controller enabled",
-			DDAv1:         newV1Agent(true),
+			Name: "v2alpha1 Admission Controller enabled",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(true).
+				Build(),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("hostip"),
+			ClusterAgent:  testBasicAdmissionController(),
 		},
-
-		//////////////////////////
-		// v2Alpha1.DatadogAgent
-		//////////////////////////
 		{
-			Name:          "v2alpha1 admission controller not enabled",
-			DDAv2:         newV2Agent(false, "", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}),
-			WantConfigure: false,
+			Name: "v2alpha1 Admission Controller enabled with full configuration",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(true).
+				WithMutateUnlabelled(true).
+				WithServiceName("testServiceName").
+				WithAgentCommunicationMode("testConfigMode").
+				WithWebhookName("testWebhookName").
+				Build(),
+			WantConfigure: true,
+			ClusterAgent:  testAdmissionControllerFull(),
 		},
 		{
-			Name:          "v2alpha1 admission controller enabled",
-			DDAv2:         newV2Agent(true, "", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}),
+			Name: "v2alpha1 Admission Controller enabled with enabled with APM uds mode",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(true).
+				WithAPMEnabled(true).
+				WithAPMUDSEnabled(true, apmSocketHostPath).
+				Build(),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources(""),
+			ClusterAgent:  testAdmissionControllerWithAPMUsingUDS(),
 		},
 		{
-			Name:          "v2alpha1 admission controller enabled, apm uses uds",
-			DDAv2:         newV2Agent(true, "", apmUDS, &v2alpha1.DogstatsdFeatureConfig{}),
+			Name: "v2alpha1 Admission Controller enabled with DSD uds mode",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(true).
+				WithDogstatsdUnixDomainSocketConfigEnabled(true).
+				WithDogstatsdUnixDomainSocketConfigPath(customPath).
+				Build(),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("socket"),
+			ClusterAgent:  testAdmissionControllerWithDSDUsingUDS(),
```

Comment:
nit - I think earlier version was a bit more readable, it seems between these two tests difference is the communication mode in DCA and it's obvious from just looking at the test. Now I have to look into both `testAdmissionControllerWithAPMUsingUDS` and `testAdmissionControllerWithDSDUsingUDS` and see the diff between multiple env vars.